ddtrace/contrib/algoliasearch/patch.py

```diff
@@ -0,0 +1,138 @@
+from importlib import import_module
+
+from ddtrace.compat import stringify
+from ddtrace.ext import AppTypes
+from ddtrace.pin import Pin
+from ddtrace.settings import config
+from ddtrace.utils.wrappers import unwrap as _u
+from ddtrace.vendor.wrapt import wrap_function_wrapper as _w
+
+DD_PATCH_ATTR = '_datadog_patch'
+
+SERVICE_NAME = 'algoliasearch'
+APP_NAME = 'algoliasearch'
+SEARCH_SPAN_TYPE = 'algoliasearch.search'
+
+# Default configuration
+config._add('algoliasearch', dict(
+    service_name=SERVICE_NAME,
+    app=APP_NAME,
+    app_type=AppTypes.db,
+    collect_query_text=False
+))
+
+
+def patch():
+    try:
+        algoliasearch = import_module('algoliasearch')
+    except ImportError:
+        return
+
+    if getattr(algoliasearch, DD_PATCH_ATTR, False):
+        return
+
+    setattr(algoliasearch, '_datadog_patch', True)
+    _w(algoliasearch.index, 'Index.search', _get_patched_search(algoliasearch))
+    Pin(
+        service=config.algoliasearch.service_name, app=config.algoliasearch.app,
+        app_type=config.algoliasearch.app_type
+    ).onto(algoliasearch.index.Index)
+
+
+def unpatch():
+    try:
+        algoliasearch = import_module('algoliasearch')
+    except ImportError:
+        return
+
+    if getattr(algoliasearch, DD_PATCH_ATTR, False):
+        setattr(algoliasearch, DD_PATCH_ATTR, False)
+        _u(algoliasearch.index.Index, 'search')
+
+
+# DEV: this maps serves the dual purpose of enumerating the algoliasearch.search() query_args that
+# will be sent along as tags, as well as converting arguments names into tag names compliant with
+# tag naming recommendations set out here: https://docs.datadoghq.com/tagging/
+QUERY_ARGS_DD_TAG_MAP = {
+    'page': 'page',
+    'hitsPerPage': 'hits_per_page',
+    'attributesToRetrieve': 'attributes_to_retrieve',
+    'attributesToHighlight': 'attributes_to_highlight',
+    'attributesToSnippet': 'attributes_to_snippet',
+    'minWordSizefor1Typo': 'min_word_size_for_1_typo',
+    'minWordSizefor2Typos': 'min_word_size_for_2_typos',
+    'getRankingInfo': 'get_ranking_info',
+    'aroundLatLng': 'around_lat_lng',
+    'numericFilters': 'numeric_filters',
+    'tagFilters': 'tag_filters',
+    'queryType': 'query_type',
+    'optionalWords': 'optional_words',
+    'distinct': 'distinct'
+}
+
+
+def _get_patched_search(algoliasearch):
+    def _search(func, instance, wrapt_args, kwargs):
```

Comment:
You can just have as `args`, or else can do `wrapt_args, wrapt_kwargs`. Having them not have the same prefix might imply that they come from different sources.